test/pages/sentinel_create_samples.spec.js

```diff
@@ -69,153 +36,148 @@ describe('lighthouse sentinel cherrypick', () => {
+              },
             },
           },
-        },
-      ]
-
-      wrapper.vm.handleSentinelSampleCreationResponse(response)
-      await wrapper.vm.$nextTick()
-
-      expect(wrapper.findComponent({ ref: 'alert' }).text()).toMatch(/an error 1, an error 2/)
-      expect(wrapper.vm.items).toEqual(
-        response
-          .slice(2)
-          .map((r) => r.data)
-          .map((r) => r.data)
-      )
-    })
-
-    it('on partial success/failure, last request failed', async () => {
-      response = [
-        {
-          errors: ['an error 2'],
-        },
-        {
-          data: {
+        ]
+        api.createSamples = vi.fn().mockReturnValue(response)
+
+        await wrapper.vm.handleSentinelSampleCreation()
+        await wrapper.vm.$nextTick()
+
+        expect(wrapper.findComponent({ ref: 'alert' }).text()).toMatch(/an error 1, an error 2/)
+        expect(wrapper.vm.items).toEqual(
+          response
+            .slice(2)
+            .map((r) => r.data)
+            .map((r) => r.data)
+        )
+      })
+
+      it('on partial success/failure, last request failed', async () => {
+        const response = [
+          {
+            errors: ['an error 2'],
+          },
+          {
             data: {
-              plate_barcode: 'aBarcode1',
-              centre: 'tst1',
-              count_fit_to_pick_samples: 1,
+              data: {
+                plate_barcode: 'aBarcode1',
+                centre: 'tst1',
+                count_fit_to_pick_samples: 1,
+              },
             },
           },
-        },
-        {
-          data: {
+          {
             data: {
-              plate_barcode: 'aBarcode2',
-              centre: 'tst1',
-              count_fit_to_pick_samples: 1,
+              data: {
+                plate_barcode: 'aBarcode2',
+                centre: 'tst1',
+                count_fit_to_pick_samples: 1,
+              },
             },
           },
-        },
-        {
-          errors: ['an error 1'],
-        },
-      ]
-
-      wrapper.vm.handleSentinelSampleCreationResponse(response)
-      await wrapper.vm.$nextTick()
-
-      expect(wrapper.findComponent({ ref: 'alert' }).text()).toMatch(
-        /Some samples were successfully created however: an error 2, an error 1/
-      )
-      expect(wrapper.vm.items).toEqual(
-        response
-          .slice(1, 3)
-          .map((r) => r.data)
-          .map((r) => r.data)
-      )
+          {
+            errors: ['an error 1'],
+          },
+        ]
+        api.createSamples = vi.fn().mockReturnValue(response)
+
+        await wrapper.vm.handleSentinelSampleCreation()
+        await wrapper.vm.$nextTick()
+
+        expect(wrapper.findComponent({ ref: 'alert' }).text()).toMatch(
+          /Some samples were successfully created however: an error 2, an error 1/
+        )
+        expect(wrapper.vm.items).toEqual(
+          response
+            .slice(1, 3)
+            .map((r) => r.data)
```

Comment:
think this double map can be .map((r) => r.data.data)

Reply:
Yes, valid point.  I hadn't noticed this pattern being used.